pkg/visibility/server.go

```diff
@@ -47,12 +47,11 @@ type server struct {
 
 // CreateAndStartVisibilityServer creates visibility server injecting KueueManager and starts it
 func CreateAndStartVisibilityServer(kueueMgr *queue.Manager, ctx context.Context) {
-	config := newVisibilityServerConfig()
```

Comment:
Are these changes needed for the fix, or drive-by cleanup? If just cleanup maybe a dedicated PR is preferred IMO.

Reply:
These changes were necessary for setting up the integration tests for the visibility API, to make it possible to reuse the API server configuration for the test client.

I've removed those integration tests altogether so these changes are not needed anymore.